WordPress/src/main/java/org/wordpress/android/ui/compose/components/menu/dropdown/MenuElementData.kt

```diff
@@ -0,0 +1,25 @@
+package org.wordpress.android.ui.compose.components.menu.dropdown
+
+import androidx.annotation.DrawableRes
+
+sealed class MenuElementData(
+    open val text: String,
+    @DrawableRes open val leadingIcon: Int,
+    open val hasDivider: Boolean,
```

Comment:
Initially I had a separate object for it, but then I considered if it's not confusing for `Divider` to have properties such as `text` and `leadingIcon`.

>I think the main advantage there is that whoever is building the menu doesn't have to think about "is this Divider gonna show up above or below the item it is part of"

Maybe if we call it `hasBottomDivider`?

This is my main concern:

<img width="289" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/592a2672-2926-4a70-8032-5cc3d0e65aaa">

Is it more confusing for the client to understand whether the divider should appear on top or bottom (even if we rename it to `hasBottomDivider`) or the fact that a `divider` instance actually has `text` and `leadingIcon` that will never be shown?

Reply:
> Is it more confusing for the client to understand whether the divider should appear on top or bottom (even if we rename it to hasBottomDivider) or the fact that a divider instance actually has text and leadingIcon that will never be shown?

I think that `text` and `leadingIcon` issue could be solved with an extra layer in the sealed class, so we have an `Item`, which could be `Individual` or `SubMenu` (naming might need tweaking), and a `Divider` as `MenuElementData`. Example:

```kotlin
sealed interface MenuElementData {
    data object Divider : MenuElementData

    sealed class Item(
        open val text: String,
        @DrawableRes open val leadingIcon: Int,
    ) : MenuElementData {
        // Item element that closes the menu when clicked
        data class Individual(
            override val text: String,
            val onClick: () -> Unit,
            @DrawableRes override val leadingIcon: Int = NO_ICON,
        ) : Item(text, leadingIcon)

        // Sub-menu element that opens a sub-menu when clicked
        data class SubMenu(
            override val text: String,
            val children: List<MenuElementData>,
            @DrawableRes override val leadingIcon: Int = NO_ICON,
        ) : Item(text, leadingIcon)
    }
}
```

I don't like the idea of the divider being an internal field on an item because it could become very unstable in terms of reusability, for example: if we have a requirement to dynamically decide which items are shown the logic becomes very complex to decide which item should have the divider since you would have to check which items are being shown in advance to define those boundaries.

Example pseudocode using Divider as a separate entity:
```kotlin
fun menuItems(showA: Boolean, showB: Boolean, showC: Boolean, showD: Boolean): List<MenuElementData> {
    val section1Shown = showA || showB || showC
    val section2Shown = showD
    return mutableListOf<MenuElementData>().apply {
        if (showA) add(Item.Individual("Section 1 - A"))
        if (showB) add(Item.SubMenu("Section 1 - B", children = listOf(...))
        if (showC) add(Item.Individual("Section 1 - C"))

        if (section1Shown && section2Shown) add(Divider)

        if (showD) add(Item.Individual("Section 2 - D"))
    }
}
```

And now without the separate entity:
```kotlin
fun menuItems(showA: Boolean, showB: Boolean, showC: Boolean, showD: Boolean): List<MenuElementData> {
    val section2Shown = showD
    return mutableListOf<MenuElementData>().apply {
        if (showA) add(Item.Individual("Section 1 - A", hasDivider = section2Shown && !showB && !showC))
        if (showB) add(Item.SubMenu("Section 1 - B", children = listOf(...), hasDivider = section2Shown && !showC))
        if (showC) add(Item.Individual("Section 1 - C", hasDivider = section2Shown))
        
        if (showD) add(Item.Individual("Section 2 - D"))
    }
}
```

Reply:
Done 👍